packages/react-native-adapter/package.json

```diff
@@ -25,7 +25,17 @@
   },
   "files": ["dist/*", "src/*"],
   "dependencies": {
-    "@walletconnect/react-native-compat": "^2.13.0"
+    "@walletconnect/react-native-compat": "^2.13.0",
+    "aws-amplify": "^5.3.18",
+    "amazon-cognito-identity-js": "^6.3.12",
+    "react-native-inappbrowser-reborn": "^3.7.0",
+    "react-native-mmkv": ">=2.11.0",
+    "react-native-aes-gcm-crypto": "^0.2.2",
+    "react-native-quick-base64": "^2.1.0",
+    "react-native-quick-crypto": "^0.6.1",
+    "@aws-sdk/client-lambda": "3.549.0",
+    "@aws-sdk/credential-providers": "3.549.0",
+    "shamir-secret-sharing": "^0.0.3"
```

Comment:
we should probably pin all of these and not allow `^` (we'll explicitly decide what to upgrade and when)

Reply:
pinned the JS libs, but actually for the native ones, there's no point adding them here right? since the consumer has to add them themselves anyways

Reply:
yeah so:
- for things consumer *has* to install additionally (native deps for linking) -> peer dep + dev dep, both with `^` version
- for things that are pure JS deps, pin them and make them direct dependencies

Reply:
yep

Reply:
so its actually I think this works

## Adapter
- hard = js deps with pinned version
- peer = mandatory native deps with `*` version (or maybe `^` if we need a particular min version)

## main package
- dev = js + native deps (with pinned version)

idea is that i only need dev deps in the main package to compile, since the idea is that to use RN you need to install the adapter AND all the adapter peers

Reply:
that's fine but there's no reason to not also list the peers as optional in the main package, feels cleaner to include them there

re `*` versions we should definitely not do that, very scary - `^` is fine for peers, we want to control "at least x version" and `^` does that (within highest set version range) so `^0.1.0` allows `0.1.x`, but not `0.2.x` or `1.x.x` etc - or with major version `^1.0.0` allows `1.x.x` but not `2.x.x` (same for patch, too technically but there any higher version will be not be allowed (since it then only acts on the patch version)

Reply:
tl;dr it should be:

### adapter
- hard = js deps with pinned version
- peer = mandatory native deps with `^`

### main
- peer = adapter (workspace version, will get auto pinned on release) + mandatory native deps with `^` (in sync with adapter) -- all of these optional
- dev = same thing as peer

Reply:
i think id rather only set the RN peer deps in the react-native-adapter - dont want bundlers to eagerly install these unless you're explicitely installing the RN adapter.

for the peer version, I basically want "anything above this version is fine" - I dont want to block ppl upgrading a minor/major if the API that we use internally is non breaking

Reply:
package managers will install dev dependencies regardless of what you do

Reply:
you overestimate the ecosystem ;)

Reply:
if that's what you want then use `>=` for the version, but never *